server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesNodeRequestTests.java

```diff
@@ -144,13 +166,41 @@ protected FieldCapabilitiesNodeRequest mutateInstance(FieldCapabilitiesNodeReque
                 return new FieldCapabilitiesNodeRequest(
                     instance.shardIds(),
                     instance.fields(),
+                    instance.filters(),
+                    instance.allowedTypes(),
                     instance.originalIndices(),
                     instance.indexFilter(),
                     instance.nowInMillis(),
                     runtimeFields
                 );
             }
-            default -> throw new IllegalStateException("The test should only allow 5 parameters mutated");
+            case 6 -> {
+                String[] randomFilter = instance.filters().length > 0 ? Strings.EMPTY_ARRAY : new String[] { "-nested" };
+                return new FieldCapabilitiesNodeRequest(
+                    instance.shardIds(),
+                    instance.fields(),
+                    randomFilter,
+                    instance.allowedTypes(),
+                    instance.originalIndices(),
+                    instance.indexFilter(),
+                    instance.nowInMillis(),
+                    instance.runtimeFields()
+                );
+            }
+            case 7 -> {
+                String[] randomType = instance.allowedTypes().length > 0 ? Strings.EMPTY_ARRAY : new String[] { "text" };
+                return new FieldCapabilitiesNodeRequest(
+                    instance.shardIds(),
+                    instance.fields(),
+                    instance.filters(),
+                    randomType,
+                    instance.originalIndices(),
+                    instance.indexFilter(),
+                    instance.nowInMillis(),
+                    instance.runtimeFields()
+                );
+            }
+            default -> throw new IllegalStateException("The test should only allow 7 parameters mutated");
```

Comment:
I think we are missing a serialization tests that uses multiple versions, to exercise the bwc layer.